packages/components/src/internal/components/labels/BarTenderSettingsForm.tsx

```diff
@@ -190,61 +189,68 @@ export const BarTenderSettingsFormImpl: FC<Props> = memo(props => {
             .catch(() => {
                 onConnectionFailure(FAILED_NOTIFICATION_MESSAGE);
             });
-    }, [api, btServiceURL]);
-
-    const isBlank = !btServiceURL || btServiceURL.trim() === '';
-
-    if (loading) return <LoadingSpinner />;
+    }, [api, btServiceURL, onConnectionFailure]);
 
     return (
         <Row>
             <Col xs={12}>
                 <Panel title={title}>
                     <Panel.Heading>{title}</Panel.Heading>
                     <Panel.Body>
-                        {connectionValidated && (
-                            <div>
-                                <Alert bsStyle="success">{SUCCESSFUL_NOTIFICATION_MESSAGE}</Alert>
-                            </div>
-                        )}
-                        {connectionValidated === false && <Alert bsStyle="danger">{failureMessage}</Alert>}
-
-                        <SettingsInput
-                            description="URL of the BarTender service to use when printing labels."
-                            label="BarTender Web Service URL"
-                            name="btServiceURL"
-                            onChange={onChangeHandler}
-                            type="url"
-                            value={btServiceURL}
-                        >
-                            <div className="pull-right">
-                                <HelpLink topic={BAR_TENDER_TOPIC} className="label-printing--help-link">
-                                    Learn more about BarTender
-                                </HelpLink>
-                            </div>
-                        </SettingsInput>
-
-                        <div className="bt-service-buttons">
-                            <SaveButton dirty={dirty} onSave={onSave} submitting={submitting} testing={testing} />
-
-                            <Button
-                                className="button-right-spacing pull-right"
-                                bsStyle="default"
-                                disabled={isBlank}
-                                onClick={onVerifyBarTenderConfiguration}
-                            >
-                                Test Connection
-                            </Button>
-                        </div>
-                        {isAppHomeFolder(container, moduleContext) && (
-                            <div className="label-templates-panel">
-                                <LabelsConfigurationPanel
-                                    {...props}
-                                    api={api}
-                                    defaultLabel={defaultLabel}
-                                    container={container}
-                                />
-                            </div>
+                        {error && <Alert>{error}</Alert>}
+                        {loading && <LoadingSpinner />}
+                        {!loading && (
```

Comment:
should it be !loading && !error?

Reply:
The default state is supported so I left the loading check the same.